homeassistant/components/zwave_js/fan.py

```diff
@@ -111,9 +120,90 @@ def percentage_step(self) -> float:
     @property
     def speed_count(self) -> int:
         """Return the number of speeds the fan supports."""
-        return int_states_in_range(SPEED_RANGE)
+        return int_states_in_range(DEFAULT_SPEED_RANGE)
 
     @property
     def supported_features(self) -> int:
         """Flag supported features."""
         return SUPPORTED_FEATURES
+
+
+class ConfiguredSpeedRangeZwaveFan(ZwaveFan):
+    """A Zwave fan with a configured speed range (e.g., 1-24 is low)."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize the fan."""
+        super().__init__(config_entry, client, info)
+        self.data_template = cast(
+            FanSpeedDataTemplate, self.info.platform_data_template
+        )
+
+    async def async_set_percentage(self, percentage: int | None) -> None:
+        """Set the speed percentage of the fan."""
+        target_value = self.get_zwave_value(TARGET_VALUE_PROPERTY)
+
+        # Entity should be unavailable if this isn't set
+        assert self.speed_configuration
+
+        if percentage is None:
+            # Value 255 tells device to return to previous value
+            zwave_speed = 255
+        elif percentage == 0:
+            zwave_speed = 0
+        else:
+            assert 0 <= percentage <= 100
+            zwave_speed = self.speed_configuration[
+                math.ceil(percentage / self.percentage_step) - 1
+            ]
+
+        await self.info.node.async_set_value(target_value, zwave_speed)
+
+    @property
+    def available(self) -> bool:
+        """Return whether the entity is available."""
+        return super().available and self.speed_configuration is not None
```

Comment:
This will change depending on the answer above.

Reply:
```suggestion
        return super().available and self.speed_configuration.value is not None
```

Reply:
These checks are now in discovery_data_template.py